File: JS/1_subarray_with_given_sum.js
const input = prompt();

const numbers = input.split(" ").map(Number);

const target = Number(prompt());

numbers.sort((a,b) => a-b);

if(target>numbers.reduce((acc,rem)=>acc+rem,0)){
    console.log("No combinations");
}
else{
    for(let i=0;i<numbers.length-1;i++){
        for(let j=i;j<numbers.length;j++){
            if(numbers[i]+numbers[j]==target){
                console.log(`[${numbers[i]}, ${numbers[j]}]`);
                break;
            }
        }
    }
}

File: Java/1_subarray_with_given_sum.java
import java.util.Scanner;
import java.util.Arrays;

class Soution{
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        String[] numbersStr = input.split(" ");    
        int[] numbers = new int[numbersStr.length];

        for(int i=0;i<numbersStr.length;i++){
            numbers[i] = Integer.parseInt(numbersStr[i]);
        }

        int target = scanner.nextInt();
        
        Array.sort(numbers);

        if(target > Arrays.stream(numbers).sum()){
            System.out.println("No combination");
            return;
        }

        else{
            for(int j=0;j<numbers.length-1;j++){
                for(int i=j+1;i<numbers.length;i++){
                   if(numbers[i]+numbers[j]==target){
                    System.out.println("["+numbers[i]+", "+numbers[j]+"]");
                    return;
                   } 
                }
            }
        }
        
    }
}